pytermgui/ansi_interface.py

```diff
@@ -107,243 +142,263 @@ class Color:
         "brightcyan": 14,
         "brightwhite": 15,
     }
-    """16 default named colors. Expanding this list will expand the names `pytermgui.parser.markup`
-    will recognize, but if that is your objective it is better to use
-    `pytermgui.parser.MarkupLanguage.alias`."""
+    """
+    16 default named colors.
 
-    def __init__(self, layer: int = 0) -> None:
+    Expanding this list will expand the names `pytermgui.parser.markup` will recognize, but if that is your objective it is better to use `pytermgui.parser.MarkupLanguage.alias`.
+    """
+
+    def __init__(self, layer: Literal[0, 1] = 0) -> None:
         """Initialize object
 
-        `layer` can be either 0 or 1. This value determines whether the instance
-        will represent foreground or background colors."""
+        `layer` can be either 0 or 1. This value determines whether the instance will represent foreground or background colors.
+        """
 
-        if layer not in [0, 1]:
-            raise NotImplementedError(f"Layer {layer} can only be one of [0, 1].")
+        if layer not in (0, 1):
+            raise NotImplementedError(f"Layer {layer} can only be one of (0, 1).")
 
         self.layer_offset = layer * 10
 
     @staticmethod
     def translate_hex(color: str) -> tuple[int, int, int]:
-        """Translate hex string of format #RRGGBB into an RGB tuple of integers"""
+        """
+        Translate hex string of format '#RGB' or '#RRGGBB' into an 'RGB' tuple of integers
+        """
+
+        if not _HEX_REGEX.match(color):
+            raise ValueError(f"{color} should be in '#RGB' or '#RRGGBB' format")
 
         if color.startswith("#"):
             color = color[1:]
 
-        rgb = []
-        for i in (0, 2, 4):
-            color_hex = color[i : i + 2]
-            rgb.append(int(color_hex, 16))
+        if len(color) == 3:
+            color = "".join(part * 2 for part in color)
 
-        return rgb[0], rgb[1], rgb[2]
+        rgb = int(color, 16)
+
+        return ((rgb & 0xFF0000) >> 16, (rgb & 0x00FF00) >> 8, (rgb & 0x0000FF))
```

Comment:
I appreciate that this is potentially a bit faster, but I think the readability of my solution is more useful over all. If there is a major improvement to speed I am fine with this change, however.